primitives/core/src/crypto.rs

```diff
@@ -918,6 +920,20 @@ impl<'a> TryFrom<&'a str> for KeyTypeId {
 	}
 }
 
+/// An identifier for a specific cryptographic algorithm used by a key pair
+#[derive(Debug, Copy, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Decode)]
+pub struct CryptoTypeId(pub [u8; 4]);
+
+/// A type alias of CryptoTypeId & a public key
+#[derive(Debug, Clone, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Decode)]
+pub struct CryptoTypePublicPair(pub CryptoTypeId, pub Vec<u8>);
+
+impl sp_std::fmt::Display for CryptoTypePublicPair {
+    fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+        write!(f, "{:#?}-{:x?}", self.0, self.1)
```

Comment:
Might be actually good to attempt to decode as string here and only fall back to hex when it fails (I mean `self.0`)

For `self.1` you should use `HexFormat` that will result in consistent formatting with the rest of the codebase.